src/GraphQLSchemaLinter.php

```diff
@@ -0,0 +1,238 @@
+        return;
+      case 'graphql-schema.comment-descriptions':
+        $this->compatibilityOptions['comment-descriptions'] = $value;
+        return;
+      case 'graphql-schema.old-implements-syntax':
+        $this->compatibilityOptions['old-implements-syntax'] = $value;
+        return;
+    }
+    return parent::setLinterConfigurationValue($key, $value);
+  }
+
+  public function getNodeBinary() {
+    return 'graphql-schema-linter';
+  }
+
+  protected function getMandatoryFlags() {
+    return array('--format=json');
+  }
+
+  protected function getDefaultFlags() {
+    $flags = array();
+
+    if (!empty($this->rules)) {
+      $flags[] = '--rules='.implode(',', $this->rules);
+    }
+    if ($this->configDirectory) {
+      $flags[] = '--config-directory='.$this->configDirectory;
+    }
+    if (!empty($this->customRulePaths)) {
+      $flags[] = '--custom-rule-paths='.implode(' ', $this->customRulePaths);
+    }
+    if (!empty($this->ignore)) {
+      $flags[] = '--ignore='.json_encode($this->ignore);
+    }
+    foreach ($this->compatibilityOptions as $flag => $enabled) {
+      if ($enabled) {
+        $flags[] = '--'.$flag;
+      }
+    }
+
+    return $flags;
+  }
+
+  protected function parseLinterOutput($path, $err, $stdout, $stderr) {
+    $messages = array();
+
+    switch($err) {
+      case 0: // If the process exits with 0 it means all rules passed.
+        break;
+      case 1: // if the process exits with 1 it means one or many rules failed.
+        /* Sample Output
+        {
+          "errors": [
+            {
+              "message": "The object type `QueryRoot` is missing a description.",
+              "location": {
+                "line": 5,
+                "column": 1,
+                "file": "schema.graphql"
+              },
+              "rule": "types-have-descriptions"
+            },
+            {
+              "message": "The field `QueryRoot.a` is missing a description.",
+              "location": {
+                "line": 6,
+                "column": 3,
+                "file": "schema.graphql"
+              },
+              "rule": "fields-have-descriptions"
+            }
+          ]
+        }
+        */
+
+        $json = json_decode($stdout, true);
+        $errors = $json['errors'];
+
+        foreach ($errors as $error) {
+          $message_parts = explode('. ', $error['message'], 2);
+          $rule = $error['rule'];
+
+          $message = new ArcanistLintMessage();
+          $message->setPath($path)
+            ->setCode($rule)
+            ->setName($this->getLinterName())
+            ->setLine($error['location']['line'])
+            ->setChar($error['location']['column'])
+            ->setDescription($error['message'])
+            ->setSeverity($this->getDefaultMessageSeverity($rule));
+          $messages[] = $message;
+        }
+        break;
+      case 2: // If the process exits with 2 it means an invalid configuration was provided.
+        break;
```

Comment:
Should we print something in this case?

Reply:
Yes — I wasn't able to reproduce this case but I will try again and see if I can print something reasonable.